src/pfe/portal/modules/User.js

```diff
@@ -746,8 +747,13 @@ module.exports = class User {
       return registrySecretList;
     }
 
-    const msg = "Failed to patch the Service Account";
+    const msg = "Failed to create the Codewind Secret and/or patch the Service Account, reverting changes to the Docker Config";
     log.error(msg);
+    
+    // Since patching the Service Account failed, we need to revert the update to the Docker Config
+    delete jsonObj.auths[address];
+    await fs.writeJson(this.dockerConfigFile, jsonObj);
```

Comment:
I'm not sure this will revert the file to the original content prior to this request. Looks like it will only remove the address from the file. If the file existed before we write to it then should we keep the contents of the file in memory? If `updateServiceAccountWithDockerRegisrySecret()` returns false we write the original content back to the file?

Reply:
Maysun noted that an error will be returned if the address already exists since update is not supported. So it's just the case where adding a new entry to an existing registry secret fails that we need to handle.